include/s3select_functions.h

```diff
@@ -790,6 +801,47 @@ struct _fn_utcnow : public base_function
   }
 };
 
+struct _fn_between : public base_function
+{
+
+  value res;
+  
+  bool operator()(bs_stmt_vec_t* args, variable* result)
+  {
+    int args_size = args->size();
+
+
+    if (args_size != 3)
+    {
+      throw base_s3select_exception("between operates on 3 expressions");//TODO FATAL
+    }
+
+    bs_stmt_vec_t::iterator iter = args->begin();
+
+    base_statement* second_expr = *iter;
+    iter++;    
+    base_statement* first_expr = *iter;
+    iter++;    
+    base_statement* main_expr = *iter;
+
+    //TODO types validations 
+    value second_expr_val = second_expr->eval();
+    value first_expr_val = first_expr->eval();
+    value main_expr_val = main_expr->eval();
+
+    if((main_expr_val > first_expr_val) && (main_expr_val < second_expr_val))
+    {
+      result->set_value((int64_t)true);
```

Comment:
we can avoid casting , by adding boolean to /set_value/ interface.

Reply:
ok, will add that